File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l, r = 0, len(heights)-1;
        m = 0;
        while(r > l):
            area = (r-l)*min(heights[r],heights[l]);
            if(area > m):
                m = area;
            if(heights[l] > heights[r]):
                r -= 1;
            else:
                l += 1;
        
        return m;




    
        

File: Data Structures & Algorithms/trapping-rain-water/submission-1.java
class Solution {
    public int trap(int[] height) {
        int i = 0;
        int j = height.length-1;
        int total = 0;
        int lMax = height[i];
        int rMax = height[j];

        while(i < j) {
            if(lMax < rMax) {
                i++;
                if(Math.min(lMax, rMax) - height[i] > 0) {
                    total += Math.min(lMax, rMax) - height[i];
                }
                if(height[i] > lMax) {
                    lMax = height[i];
                }
            } else {
                j--;
                if(Math.min(lMax, rMax) - height[j] > 0) {
                total += Math.min(lMax, rMax) - height[j];
            }
                if(height[j] > rMax) {
                    rMax = height[j];
                }
            }
        }
        return total;
    }
}
